### ISSUE TITLE
DLL does not export any of the new methods added to libbluray

### ISSUE THREAD

**Issue by roromo** [ShiftMediaProject/libbluray#6]
## Context

Libbluray added new methods over its development. However, the DLLs packed in the release archives of this project don't expose any of these. It's exporting the same 97 methods that were the only ones available a long time ago.

For example, the x64 DLL in the version 1.2.0 archive exports only 97 functions, instead of all 138 functions.

An example of a method that is not exported: "int bd_set_rate(BLURAY *bd, uint32_t rate)" from bluray.h

Expected exports for v1.2.0:
Java_java_awt_BDFontMetrics_charWidthN
Java_java_awt_BDFontMetrics_charsWidthN
Java_java_awt_BDFontMetrics_destroyFontN
Java_java_awt_BDFontMetrics_destroyN
Java_java_awt_BDFontMetrics_getFontFamilyAndStyleN
Java_java_awt_BDFontMetrics_initN
Java_java_awt_BDFontMetrics_loadFontN
Java_java_awt_BDFontMetrics_resolveFontN
Java_java_awt_BDFontMetrics_stringWidthN
Java_java_awt_BDFontMetrics_unloadFontConfigN
Java_java_awt_BDGraphics_drawStringN
Java_org_videolan_Libbluray_cacheBdRomFileN
Java_org_videolan_Libbluray_getAacsDataN
Java_org_videolan_Libbluray_getBdjoN
Java_org_videolan_Libbluray_getPlaylistInfoN
Java_org_videolan_Libbluray_getTitleInfosN
Java_org_videolan_Libbluray_getUOMaskN
Java_org_videolan_Libbluray_listBdFilesN
Java_org_videolan_Libbluray_readRegN
Java_org_videolan_Libbluray_seekN
Java_org_videolan_Libbluray_selectAngleN
Java_org_videolan_Libbluray_selectPlaylistN
Java_org_videolan_Libbluray_selectRateN
Java_org_videolan_Libbluray_selectTitleN
Java_org_videolan_Libbluray_setKeyInterestN
Java_org_videolan_Libbluray_setUOMaskN
Java_org_videolan_Libbluray_setVirtualPackageN
Java_org_videolan_Libbluray_soundEffectN
Java_org_videolan_Libbluray_tellTimeN
Java_org_videolan_Libbluray_updateGraphicN
Java_org_videolan_Libbluray_writeRegN
Java_org_videolan_Logger_logN
bd_bdj_osd_cb
bd_bdj_seek
bd_bdj_sound_effect
bd_chapter_pos
bd_close
bd_debug
bd_free_bdjo
bd_free_clpi
bd_free_mobj
bd_free_mpls
bd_free_title_info
bd_get_aacs_data
bd_get_clpi
bd_get_current_angle
bd_get_current_chapter
bd_get_current_title
bd_get_debug_mask
bd_get_disc
bd_get_disc_info
bd_get_event
bd_get_main_title
bd_get_meta
bd_get_meta_file
bd_get_playlist_info
bd_get_scr
bd_get_sound_effect
bd_get_title_info
bd_get_title_size
bd_get_titles
bd_get_uo_mask
bd_get_version
bd_gpr_read
bd_gpr_write
bd_init
bd_lock_osd_buffer
bd_menu_call
bd_mouse_select
bd_mutex_destroy
bd_mutex_init
bd_mutex_lock
bd_mutex_unlock
bd_open
bd_open_dir
bd_open_disc
bd_open_file_dec
bd_open_files
bd_open_stream
bd_play
bd_play_playlist_at
bd_play_title
bd_play_title_internal
bd_psr_lock
bd_psr_read
bd_psr_register_cb
bd_psr_reset_backup_registers
bd_psr_restore_state
bd_psr_save_state
bd_psr_setting_write
bd_psr_unlock
bd_psr_unregister_cb
bd_psr_write
bd_psr_write_bits
bd_read
bd_read_bdjo
bd_read_clpi
bd_read_ext
bd_read_file
bd_read_mobj
bd_read_mpls
bd_read_skip_still
bd_refcnt_dec
bd_refcnt_inc
bd_reg_read
bd_reg_write
bd_register_argb_overlay_proc
bd_register_dir
bd_register_file
bd_register_overlay_proc
bd_registers_free
bd_registers_init
bd_seamless_angle_change
bd_seek
bd_seek_chapter
bd_seek_mark
bd_seek_playitem
bd_seek_time
bd_select_angle
bd_select_playlist
bd_select_rate
bd_select_stream
bd_select_title
bd_set_bdj_kit
bd_set_bdj_uo_mask
bd_set_debug_handler
bd_set_debug_mask
bd_set_player_setting
bd_set_player_setting_str
bd_set_rate
bd_set_scr
bd_set_virtual_package
bd_start_bdj
bd_stop_bdj
bd_tell
bd_tell_time
bd_unlock_osd_buffer
bd_user_input

Exports of the x64 DLL in the 1.2.0 release archive:
Java_java_awt_BDFontMetrics_charWidthN
Java_java_awt_BDFontMetrics_charsWidthN
Java_java_awt_BDFontMetrics_destroyFontN
Java_java_awt_BDFontMetrics_destroyN
Java_java_awt_BDFontMetrics_getFontFamilyAndStyleN
Java_java_awt_BDFontMetrics_initN
Java_java_awt_BDFontMetrics_loadFontN
Java_java_awt_BDFontMetrics_resolveFontN
Java_java_awt_BDFontMetrics_stringWidthN
Java_java_awt_BDFontMetrics_unloadFontConfigN
Java_java_awt_BDGraphics_drawStringN
Java_org_videolan_Libbluray_cacheBdRomFileN
Java_org_videolan_Libbluray_getAacsDataN
Java_org_videolan_Libbluray_getBdjoN
Java_org_videolan_Libbluray_getPlaylistInfoN
Java_org_videolan_Libbluray_getTitleInfosN
Java_org_videolan_Libbluray_getUOMaskN
Java_org_videolan_Libbluray_listBdFilesN
Java_org_videolan_Libbluray_readRegN
Java_org_videolan_Libbluray_seekN
Java_org_videolan_Libbluray_selectAngleN
Java_org_videolan_Libbluray_selectPlaylistN
Java_org_videolan_Libbluray_selectRateN
Java_org_videolan_Libbluray_selectTitleN
Java_org_videolan_Libbluray_setKeyInterestN
Java_org_videolan_Libbluray_setUOMaskN
Java_org_videolan_Libbluray_setVirtualPackageN
Java_org_videolan_Libbluray_soundEffectN
Java_org_videolan_Libbluray_tellTimeN
Java_org_videolan_Libbluray_updateGraphicN
Java_org_videolan_Libbluray_writeRegN
Java_org_videolan_Logger_logN
bd_chapter_pos
bd_close
bd_free_bdjo
bd_free_clpi
bd_free_mobj
bd_free_mpls
bd_free_title_info
bd_get_clpi
bd_get_current_angle
bd_get_current_chapter
bd_get_current_title
bd_get_debug_mask
bd_get_disc_info
bd_get_event
bd_get_main_title
bd_get_meta
bd_get_meta_file
bd_get_playlist_info
bd_get_sound_effect
bd_get_title_info
bd_get_title_size
bd_get_titles
bd_get_version
bd_init
bd_menu_call
bd_mouse_select
bd_open
bd_open_disc
bd_open_stream
bd_play
bd_play_title
bd_read
bd_read_bdjo
bd_read_clpi
bd_read_ext
bd_read_file
bd_read_mobj
bd_read_mpls
bd_read_skip_still
bd_refcnt_dec
bd_refcnt_inc
bd_register_argb_overlay_proc
bd_register_dir
bd_register_file
bd_register_overlay_proc
bd_seamless_angle_change
bd_seek
bd_seek_chapter
bd_seek_mark
bd_seek_playitem
bd_seek_time
bd_select_angle
bd_select_playlist
bd_select_stream
bd_select_title
bd_set_debug_handler
bd_set_debug_mask
bd_set_player_setting
bd_set_player_setting_str
bd_set_scr
bd_start_bdj
bd_stop_bdj
bd_tell
bd_tell_time
bd_user_input

**Comment by Sibras** [comment]
Ive checked the dll exports and there were some missing functions. Although there were only 5 that were missing. That list of functions you provided includes functions that should not actually be exported (i.e. they are from bluray-internal.h which is only for internal use) so im not sure where you got that list from.

I have however gone through all the public headers and made sure there are no missing functions. So try the latest build and see how that goes.

**Comment by roromo** [comment]
Hi, that's the list of exported functions of the libbluray-2.dll built using MinGW x64 under MSYS2.

I used this tool to create the list of exports: https://github.com/lucasg/Dependencies

I did not check to see where all those functions were from. It means that even private functions are exported by default when building using MinGW

Thank you for looking into the issue.

I'll try tomorrow to make a build and see what happens. :)
